core/mm/vm.c

```diff
@@ -657,7 +665,10 @@ TEE_Result vm_remap(struct user_mode_ctx *uctx, vaddr_t *new_va, vaddr_t old_va,
 	fobj_put(fobj);
 
 	vm_set_ctx(uctx->ts_ctx);
-	*new_va = r_first->va;
+
+	res = PUT_USER(r_first->va, new_va);
+	if (res)
+		goto err_restore_map;
```

Comment:
I don't think `err_restore_map` will work at this stage as `fobj` is lost.
I would rather swap the above instructions:
```
	res = PUT_USER(r_first->va, new_va);
	if (res)
		goto err_restore_map;

	fobj_put(fobj);

	vm_set_ctx(uctx->ts_ctx);

Reply:
thank you for the catch. updated